packages/clerk-js/src/ui/utils/colors/modern.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Modern CSS-based color manipulation utilities
+ * Uses color-mix() and relative color syntax when supported
+ */
+
+import { cssSupports } from '../cssSupports';
+import { hasModernColorSupport as hasModernColorSupportCached } from './cache';
+
+/**
+ * Modern CSS-based color manipulation utilities
+ * Uses color-mix() and relative color syntax when supported
+ */
+export const modernColors = {
+  /**
+   * Lightens a color by a percentage using modern CSS
+   */
+  lighten: (color: string | undefined, percentage = 0): string | undefined => {
+    if (!color) return undefined;
+
+    if (cssSupports.relativeColorSyntax()) {
+      // Use relative color syntax for precise lightness control
+      const lightnessIncrease = percentage * 100; // Convert to percentage
+      return `hsl(from ${color} h s calc(l + ${lightnessIncrease}%))`;
+    }
+
+    if (cssSupports.colorMix()) {
+      // Use color-mix as fallback
+      const mixPercentage = Math.min(percentage * 100, 95); // Cap at 95%
+      return `color-mix(in srgb, ${color}, white ${mixPercentage}%)`;
+    }
+
+    return color; // Return original if no modern CSS support
```

Comment:
should we maybe throw an error if you attempt to call `modernColors.lighten()` on a browser that doesn't support color-mix or relative color?